castor-service/src/test/java/io/carbynestack/castor/service/persistence/cache/ReservationCachingServiceTest.java

```diff
@@ -58,282 +73,434 @@ public void setUp() {
+        reservationCachingService.getUnlockedReservation(reservationId, tupleType, tupleCount));
+  }
+
+  @Test
+  public void givenSuccessfulRequest_whenForgetReservation_thenCallDeleteOnCache() {
+    String reservationId = "reservationId";
+
+    reservationCachingService.forgetReservation(reservationId);
+
+    verify(redisTemplateMock).delete(testCachePrefix + reservationId);
+  }
+
+  @Test
+  public void givenAlreadyAReservationWithGivenIdInCache_whenKeepAndApplyReservation_thenThrow() {
+    String reservationId = "testReservation";
+    Reservation existingReservationMock = mock(Reservation.class);
+
+    when(existingReservationMock.getReservationId()).thenReturn(reservationId);
+    when(valueOperationsMock.get(testCachePrefix + reservationId))
+        .thenReturn(existingReservationMock);
 
     CastorServiceException actualCse =
         assertThrows(
             CastorServiceException.class,
-            () -> reservationCachingService.keepReservation(reservationMock));
+            () -> reservationCachingService.keepAndApplyReservation(existingReservationMock));
 
     assertEquals(
-        String.format(TUPLE_TYPE_MISMATCH_EXCEPTION_MSG, actualTupleType, requestedTupleType),
-        actualCse.getMessage());
+        String.format(RESERVATION_CONFLICT_EXCEPTION_MSG, reservationId), actualCse.getMessage());
   }
 
   @Test
-  public void
-      givenTuplesOfReferencedChunkNotAvailable_whenKeepReservation_throwCastorServiceException() {
-    UUID chunkId = UUID.fromString("b7b010e0-362b-401c-9560-4cf4b2a68139");
-    TupleType tupleType = TupleType.INPUT_MASK_GFP;
-    long reservedMarker = 5;
-    long startIndex = 0;
-    ReservationElement reservationElementMock = mock(ReservationElement.class);
-    Reservation reservationMock = mock(Reservation.class);
-    TupleChunkMetaDataEntity metaDataMock = mock(TupleChunkMetaDataEntity.class);
-
-    when(reservationElementMock.getTupleChunkId()).thenReturn(chunkId);
-    when(reservationElementMock.getStartIndex()).thenReturn(startIndex);
-    when(reservationMock.getTupleType()).thenReturn(tupleType);
-    when(reservationMock.getReservations()).thenReturn(singletonList(reservationElementMock));
-    when(tupleChunkMetaDataStorageServiceMock.getTupleChunkData(chunkId)).thenReturn(metaDataMock);
-    when(metaDataMock.getStatus()).thenReturn(ActivationStatus.UNLOCKED);
-    when(metaDataMock.getTupleType()).thenReturn(tupleType);
-    when(metaDataMock.getReservedMarker()).thenReturn(reservedMarker);
+  public void givenNoFragmentWithReferencedStartIndex_whenKeepAndApplyReservation_thenThrow() {
+    UUID tupleChunkId = UUID.fromString("3fd7eaf7-cda3-4384-8d86-2c43450cbe63");
+    long requestedStartIndex = 42;
+    long requestedLength = 21;
+    ReservationElement re =
+        new ReservationElement(tupleChunkId, requestedLength, requestedStartIndex);
+    String reservationId = "testReservation";
+    TupleType tupleType = MULTIPLICATION_TRIPLE_GFP;
```

Comment:
Used on only once in remainder of body -> redundant

Reply:
After looking through the whole PR, I see that this is a pattern. Set all values relevant at the beginning then use them throughout the test. I see the rationale behind it. I leave it up to you to decide whether to keep this style or not. I think it helps by adding structure on the one hand. On the other hand it generates warnings in the IDE.

Reply:
Did it for the structure but have observed the warnings too. Happy to discuss which way to go

Reply:
Let's stick with the way you did it and ignore the warnings.